aten/src/ATen/native/cpu/MultinomialKernel.cpp

```diff
@@ -0,0 +1,135 @@
+#include <ATen/ATen.h>
+
+#include <ATen/Dispatch.h>
+#include <ATen/native/Copy.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cpu/Loops.h>
+#include <ATen/core/DistributionsHelper.h>
+
+namespace at {
+namespace native {
+namespace {
+
+template<typename scalar_t>
+void multinomial_apply(Tensor& self, const Tensor& prob_dist, const int64_t n_sample, const bool with_replacement, Generator* generator) {
```

Comment:
wait, so outside of this function self is prob_dist, but then instead self is result?  That seems unnecessarily confusing...

Reply:
Renamed to result and self